stackslib/src/chainstate/stacks/db/blocks.rs

```diff
@@ -4152,23 +4153,37 @@ impl StacksChainState {
                 burn_header_hash,
                 ..
             } = &stack_stx_op;
+
+            let mut args = vec![
+                Value::UInt(*stacked_ustx),
+                // this .expect() should be unreachable since we coerce the hash mode when
+                // we parse the StackStxOp from a burnchain transaction
+                reward_addr
+                    .as_clarity_tuple()
+                    .expect("FATAL: stack-stx operation has no hash mode")
+                    .into(),
+                Value::UInt(u128::from(*block_height)),
+                Value::UInt(u128::from(*num_cycles)),
+            ];
+            // Appending additional signer related arguments for pox-4
+            if POX_4_NAME == active_pox_contract {
```

Comment:
We should start using the New and Improved `PoxVersions` enum:

```suggestion
            if active_pox_contract == PoxVersions::Pox4.get_name() {
```